File: src/mini_soar/core/remediation.py
from datetime import datetime

from pydantic import BaseModel


class RemediationRecord(BaseModel):
    id: int
    event_id: str
    event_type: str
    host: str
    service: str
    action: str
    status: str
    duration_seconds: float
    message: str | None = None
    created_at: datetime


class RemediationListResponse(BaseModel):
    count: int
    items: list[RemediationRecord]

class RemediationSummary(BaseModel):
    total: int
    success: int
    failed: int
    error: int
    skipped: int
    success_rate: float
    average_duration_seconds: float
class RemediationDistribution(BaseModel):
    status: dict[str, int]
    event_type: dict[str, int]

File: frontend/src/types/remediation.ts
export interface RemediationSummary {
  total: number;
  success: number;
  failed: number;
  error: number;
  skipped: number;
  success_rate: number;
  average_duration_seconds: number;
}

export interface RemediationRecord {
  id: number;
  event_id: string;
  event_type: string;
  host: string;
  service: string;
  action: string;
  status: string;
  duration_seconds: number;
  message: string | null;
  created_at: string;
}

export interface RemediationListResponse {
  count: number;
  items: RemediationRecord[];
}
